net/WebSocketHandler.hpp

```diff
@@ -1002,6 +1009,18 @@ class WebSocketHandler : public ProtocolHandlerInterface
         LOG_INF("WebSocket version: " << wsVersion << ", key: [" << wsKey << "], protocol: ["
                                       << wsProtocol << ']');
 
+        /* SHOULD verify the Origin field is an origin they expect. If the origin indicated is
+         * unacceptable to the server, then it SHOULD respond ... with a reply containing HTTP
+         * 403 Forbidden status code.
+         */
+        const std::string origin = req.get("Origin", "");
```

Comment:
This adds a hard requirement to have the Origin set. So the question is whether or not we have legitimate cases where the Origin might be missing. Reading [the docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Origin#description), nothing stands out.

Still, I wonder if we should put this behind a config that admins can disable (at their own risk). Conversely, we'll hear about breaks after upgrades, if they exist. So perhaps we can wait and see.

Reply:
https://github.com/CollaboraOnline/online/issues/12095 as suspected there is some breakage.
But we can problably make our origin more tolerant to https not specifying the 443 port.